integration-test-groups/azure/azure-eventhubs/src/test/java/org/apache/camel/quarkus/component/azure/eventhubs/it/AzureEventhubsTest.java

```diff
@@ -72,9 +74,10 @@ void produceConsumeEvents() {
                         .body(
                                 "body", is(messageBody),
                                 "headers.CamelAzureEventHubsEnqueuedTime", notNullValue(),
-                                "headers.CamelAzureEventHubsOffset", greaterThanOrEqualTo(0),
```

Comment:
@llowinge looks there is a strange issue here :)
What is the error message please ? Is it a known bug in hamcrest matchers ?

By any chance, would using something like `greaterThanOrEqualTo(0L)` or `greaterThanOrEqualTo(Long.valueOf(0L))` work ?

Reply:
The problem is, that camel can return through Azure Long or Integer class. With real instance, it returns Long and the test fails. Unfortunately your suggestions don't work, because it fails with typing Integer to Long class (which is not possible). I had to do it this way, to support both Integer and Long classes.

Reply:
It is a bad situation indeed. It would need kind of real instance only assertion or something like asserting that it matches any of a list of matchers.

Fair enough with the solution and many thanks for sharing :+1: